To: vim_dev@googlegroups.com
Subject: Patch 7.3.414
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.414
Problem:    Using CTRL-A on "000" drops the leading zero, while on "001" it
	    doesn't.
Solution:   Detect "000" as an octal number. (James McCoy)
Files:	    src/charset.c


*** ../vim-7.3.413/src/charset.c	2012-01-10 22:26:12.000000000 +0100
--- src/charset.c	2012-01-26 13:33:45.000000000 +0100
***************
*** 764,770 ****
  }
  
  /*
!  * Return the number of characters string "s" will take on the screen,
   * counting TABs as two characters: "^I".
   */
      int
--- 764,770 ----
  }
  
  /*
!  * Return the number of character cells string "s" will take on the screen,
   * counting TABs as two characters: "^I".
   */
      int
***************
*** 775,782 ****
  }
  
  /*
!  * Return the number of characters string "s[len]" will take on the screen,
!  * counting TABs as two characters: "^I".
   */
      int
  vim_strnsize(s, len)
--- 775,782 ----
  }
  
  /*
!  * Return the number of character cells string "s[len]" will take on the
!  * screen, counting TABs as two characters: "^I".
   */
      int
  vim_strnsize(s, len)
***************
*** 1830,1836 ****
  			hex = 0;	/* can't be octal */
  			break;
  		    }
! 		    if (ptr[n] > '0')
  			hex = '0';	/* assume octal */
  		}
  	    }
--- 1830,1836 ----
  			hex = 0;	/* can't be octal */
  			break;
  		    }
! 		    if (ptr[n] >= '0')
  			hex = '0';	/* assume octal */
  		}
  	    }
*** ../vim-7.3.413/src/version.c	2012-01-26 13:01:54.000000000 +0100
--- src/version.c	2012-01-26 13:38:04.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     414,
  /**/

-- 
No engineer can take a shower without wondering if some sort of Teflon coating
would make showering unnecessary.
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
